lib/user-methods.js

```diff
@@ -609,10 +515,64 @@ const instanceMethods = {
     },
 };
 
+const credentialsMethods = {
+    usernamePassword(username, password, createUser) {
+        checkTypes(arguments, ['string', 'string', 'boolean']);
+        return new Credentials('password', username, { register: createUser, password });
+    },
+
+    facebook(token) {
+        checkTypes(arguments, ['string']);
+        return new Credentials('facebook', token);
+    },
+
+    google(token) {
+        checkTypes(arguments, ['string']);
+        return new Credentials('google', token);
+    },
+
+    anonymous() {
+        return new Credentials('anonymous');
+    },
+
+    nickname(value, isAdmin) {
+        return new Credentials('nickname', value, { is_admin: isAdmin || false });
+    },
+
+    azureAD(token) {
+        return new Credentials('azuread', token)
+    },
+
+    jwt(token) {
+        return new Credentials('jwt', token);
+    },
+
+    adminToken(token) {
+        return new Credentials('adminToken', token);
+    },
```

Comment:
I guess we could use `checkTypes` for `nickname`, `azureAD`, `jwt` and `adminToken` as well?